tests/test.rs

```diff
@@ -43,12 +43,12 @@ fn old_const_fn_ctor() {
     const EMPTY: SmolStr = SmolStr::new_inline_from_ascii(0, b"");
     const A: SmolStr = SmolStr::new_inline_from_ascii(1, b"A");
     const HELLO: SmolStr = SmolStr::new_inline_from_ascii(5, b"HELLO");
-    const LONG: SmolStr = SmolStr::new_inline_from_ascii(22, b"ABCDEFGHIZKLMNOPQRSTUV");
+    const LONG: SmolStr = SmolStr::new_inline_from_ascii(23, b"ABCDEFGHIZKLMNOPQRSTUVW");
 
     assert_eq!(EMPTY, SmolStr::from(""));
     assert_eq!(A, SmolStr::from("A"));
     assert_eq!(HELLO, SmolStr::from("HELLO"));
-    assert_eq!(LONG, SmolStr::from("ABCDEFGHIZKLMNOPQRSTUV"));
+    assert_eq!(LONG, SmolStr::from("ABCDEFGHIZKLMNOPQRSTUVW"));
 }
 
 fn check_props(std_str: &str, smol: SmolStr) -> Result<(), proptest::test_runner::TestCaseError> {
```

Comment:
This has a heap alloc check on lines 58/59, that should have the len check adjusted as well